planner/core/runtime_filter_generator.go

```diff
@@ -0,0 +1,247 @@
+				logutil.BgLogger().Debug("Now we don't support global RF. Remove it",
+					zap.Int("BuildNodeId", runtimeFilter.buildNode.id),
+					zap.Int("TargetNodeId", physicalTableScan.id))
+				continue
+			}
+			runtimeFilter.rfMode = rfMode
+
+			// assign rf to current node
+			runtimeFilter.assign(physicalTableScan, scanOutputColumn)
+			currentRFList = append(currentRFList, runtimeFilter)
+		}
+	}
+
+	if len(currentRFList) == 0 {
+		return
+	}
+
+	// todo Since the feature of adding filter operators has not yet been implemented,
+	// the following code for this function will not be used for now.
+	// supply selection if there is no predicates above target scan node
+	//if parent, ok := generator.parentPhysicalPlan.(*PhysicalSelection); !ok {
+	//	// StatsInfo: Just set a placeholder value here, and this value will not be used in subsequent optimizations
+	//	sel := PhysicalSelection{hasRFConditions: true}.Init(plan.SCtx(), plan.statsInfo(), plan.SelectBlockOffset())
+	//	sel.fromDataSource = true
+	//	sel.SetChildren(plan)
+	//	generator.parentPhysicalPlan.SetChild(generator.childIdxForParentPhysicalPlan, sel)
+	//} else {
+	//	parent.hasRFConditions = true
+	//}
+
+	// todo
+	// filter predicate selectivity, A scan node does not need many RFs, and the same column does not need many RFs
+}
+
+func (generator *RuntimeFilterGenerator) matchRFJoinType(hashJoinPlan *PhysicalHashJoin) bool {
+	if hashJoinPlan.RightIsBuildSide() {
+		// case1: build side is on the right
+		if hashJoinPlan.JoinType == LeftOuterJoin || hashJoinPlan.JoinType == AntiSemiJoin ||
+			hashJoinPlan.JoinType == LeftOuterSemiJoin || hashJoinPlan.JoinType == AntiLeftOuterSemiJoin {
+			logutil.BgLogger().Debug("Join type does not match RF pattern when build side is on the right",
+				zap.Int32("PlanNodeId", int32(hashJoinPlan.id)),
+				zap.String("JoinType", hashJoinPlan.JoinType.String()))
+			return false
+		}
+	} else {
+		// case2: build side is on the left
+		if hashJoinPlan.JoinType == RightOuterJoin {
+			logutil.BgLogger().Debug("Join type does not match RF pattern when build side is on the left",
+				zap.Int32("PlanNodeId", int32(hashJoinPlan.id)),
+				zap.String("JoinType", hashJoinPlan.JoinType.String()))
+			return false
+		}
+	}
+	return true
+}
+
+func (generator *RuntimeFilterGenerator) matchEQPredicate(eqPredicate *expression.ScalarFunction,
+	rightIsBuildSide bool) bool {
+	var targetColumn, srcColumn *expression.Column
+	if rightIsBuildSide {
+		targetColumn = eqPredicate.GetArgs()[0].(*expression.Column)
+		srcColumn = eqPredicate.GetArgs()[1].(*expression.Column)
+	} else {
+		targetColumn = eqPredicate.GetArgs()[1].(*expression.Column)
+		srcColumn = eqPredicate.GetArgs()[0].(*expression.Column)
+	}
+	// match target column
+	// condition1: the target column must be real column
+	// condition2: the target column has not undergone any transformation
+	// todo: cast expr in target column
+	if targetColumn.IsHidden || targetColumn.OrigName == "" {
```

Comment:
I'm a bit suspecting that if `targetColumn.OrigName == ""` is a reliable check.

Reply:
I checked the current code, as long as the column in the original table is currently added to this property. So here is my judgment first. If there is a problem later, it can be repaired in time.